ipu.py

```diff
@@ -442,15 +442,37 @@ def start(self) -> None:
     def cold_boot(self) -> None:
         pass
 
-    def version(self) -> str:
+    def _version_via_redfish(self) -> Optional[str]:
+        url = f"https://{self.url}:8443/redfish/v1/Managers/1"
+        res = self._run_curl(f"-k '{url}'")
+        if res.returncode != 0:
+            return None
+        try:
+            fwversion = (json.loads(res.out))["FirmwareVersion"]
```

Comment:
why `()` around `json.loads`

also, simplify this a bit further?

```
j = json.load(res.out)
if "FirmwareVersion" not in j
```

Otherwise, the PR lgtm.

Reply:
changed.